kmir/src/kmir/kdist/mir-semantics/symbolic/kmir-symbolic-locals.md

```diff
@@ -0,0 +1,98 @@
+              ),
+              FUNCTIONNAMES
+            )
+        =>
+           #reserveSymbolicsFor(LOCALS, ARGCOUNT)
+        ~> #execBlock(FIRST)
+         ...
+       </k>
+       <currentFunc> _ => #tyFromName(SYMNAME, FUNCTIONNAMES) </currentFunc>
+       <currentFrame>
+         <currentBody> _ => toKList(BLOCKS) </currentBody>
+         <caller> _ => ty(-1) </caller> // no caller
+         <dest> _ => place(local(-1), .ProjectionElems)</dest>
+         <target> _ => noBasicBlockIdx </target>
+         <unwind> _ => unwindActionUnreachable </unwind>
+         <locals> _ => #reserveFor(RETURNLOCAL) </locals>
+       </currentFrame>
+```
+
+## Declare symbolic arguments based on their types
+
+```k
+  syntax KItem ::= #reserveSymbolicsFor( LocalDecls, Int )
+
+  rule <k> #reserveSymbolicsFor( .LocalDecls, _ ) => .K ... </k>
+
+  rule <k> #reserveSymbolicsFor( LOCALS:LocalDecls, 0 ) => .K ... </k>
+       <locals> ... .List => #reserveFor(LOCALS) </locals> // No more arguments, treat the rest of the Decls normally
+
+```
+
+## Integers
+
+```k
+  // Unsigned
+  rule <k> #reserveSymbolicsFor( localDecl(TY, _, MUT) LOCALS:LocalDecls, COUNT        )
+        => #reserveSymbolicsFor(                       LOCALS:LocalDecls, COUNT -Int 1 )
+           ...
+       </k>
+       <locals> ... .List => ListItem(typedValue( Integer(?INT:Int, #bitWidth(PRIMTY), false), TY, MUT )) </locals>
+       <types> ... TY |-> typeInfoPrimitiveType ( primTypeUint( PRIMTY ) ) ... </types>
+    requires 0 <Int COUNT
+    ensures #intConstraints( ?INT, PRIMTY )
+
+  // Signed
+  rule <k> #reserveSymbolicsFor( localDecl(TY, _, MUT) LOCALS:LocalDecls, COUNT        )
+        => #reserveSymbolicsFor(                       LOCALS:LocalDecls, COUNT -Int 1 )
+           ...
+       </k>
+       <locals> ... .List => ListItem(typedValue( Integer(?INT:Int, #bitWidth(PRIMTY), true), TY, MUT )) </locals>
+       <types> ... TY |-> typeInfoPrimitiveType ( primTypeInt( PRIMTY ) ) ... </types>
+    requires 0 <Int COUNT
+    ensures #intConstraints( ?INT, PRIMTY )
+
+  syntax Bool ::= #intConstraints( Int, InTy ) [function, total]
+
+  rule #intConstraints( X, TY:IntTy  ) => 0 -Int (2 ^Int (#bitWidth(TY) -Int 1)) <=Int X andBool X <Int 2 ^Int (#bitWidth(TY) -Int 1)
+  rule #intConstraints( X, TY:UintTy ) => 0                                      <=Int X andBool X <Int 2 ^Int  #bitWidth(TY)
+```
+
+## Boolean Values
+
+```k
+  rule <k> #reserveSymbolicsFor( localDecl(TY, _, MUT) LOCALS:LocalDecls, COUNT        )
+        => #reserveSymbolicsFor(                       LOCALS:LocalDecls, COUNT -Int 1 )
+           ...
+       </k>
+       <locals> ... .List => ListItem(typedValue( BoolVal( ?_BOOL:Bool ), TY, MUT )) </locals>
+       <types> ... TY |-> typeInfoPrimitiveType ( primTypeBool ) ... </types>
+    requires 0 <Int COUNT
+
+endmodule
```

Comment:
We also need a rule for arguments which aren't `Int` or `Bool` valued, they would just be symbolic altogether.